narwhal/types/src/failpoints.rs

```diff
@@ -0,0 +1,42 @@
+use std::env::{self, VarError};
+use tracing::{info, warn};
+
+pub fn initialize_failpoints() {
+    if fail::has_failpoints() {
+        warn!("Failpoints are enabled");
+        let failpoints = match env::var("FAILPOINTS") {
```

Comment:
We also need to have a mechanism that completely has failpoints disabled at compile time in production binaries, potentially via cargo feature flags. It seems like the `fail::has_failpoints()` call is intended to do that based on a feature flag to the fail crate, but i don't see how that feature flag is expected to be manipulated in this PR.

Reply:
Hmm failpoints are guarded by a feature flag, `--features fail/failpoints`. So by default it is disabled. This is a [sample workflow](https://github.com/MystenLabs/sui-operations/blob/e36bcfe8bf8b33ed0075b051a7f1fcc1780d8de2/.github/workflows/sui-release-binaries-with-failpoints.yml#L56) for how I deploy it in a test environment. Are you saying there should be a way to disable it even if the feature flag is set? (if so can you give me an example of how this is done 🙏🏽 )

Reply:
I think this future flag should be sufficient